### ISSUE TITLE
Early stop with multiple validation sets

### ISSUE THREAD

**Issue by changyc14** [Microsoft/LightGBM#1168]
Hi,

If I have more than 1 validation set to be fed into the parameter "valids", and I use early stop, how does my model determine when to stop? In other words, what best score does it keep? Is it based on the average score of all my validation sets?  Or just on particular set? When I reviewed source code of cb.early.stop (https://github.com/Microsoft/LightGBM/blob/master/R-package/R/callback.R), I was not able to figure it out. I made a small snippet of code to describe the condition above. Really appreciate it if anyone is willing to provide me with leads. Thank you.

bst <- lgb.train(params, dtrain, nrounds = MAX_ROUNDS ,valids = list(train = dtrain, test1 = dval_1, test2 = dval_2), early_stopping_rounds = 30, verbose = 1)

**Comment by Laurae2** [comment]
The last validation set defines the early stopping.

Reproducible example:

```r
library(lightgbm)
data(agaricus.train, package = "lightgbm")
train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label = train$label)
data(agaricus.test, package = "lightgbm")
test <- agaricus.test
test$label[1000] <- 0
dtest1 <- lgb.Dataset.create.valid(dtrain, test$data[1:500,], label = test$label[1:500])
dtest2 <- lgb.Dataset.create.valid(dtrain, test$data[501:1000, ], label = test$label[501:1000])
params <- list(objective = "regression", metric = "l2")
valids <- list(test1 = dtest1, test2 = dtest2)
model <- lgb.train(params,
                   dtrain,
                   100,
                   valids,
                   min_data = 1,
                   learning_rate = 0.99,
                   early_stopping_rounds = 2)
```

Yields:

```r
[LightGBM] [Info] Total Bins 137
[LightGBM] [Info] Number of data: 6513, number of used features: 116
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[1]:	test1's l2:2.36725e-05	test2's l2:0.0020044 
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[2]:	test1's l2:2.36726e-09	test2's l2:0.0019998 
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[3]:	test1's l2:2.36727e-13	test2's l2:0.002 
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[4]:	test1's l2:2.36729e-17	test2's l2:0.002 
```
